pkg/volume/util/atomic_writer.go

```diff
@@ -0,0 +1,305 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	utilerrors "k8s.io/kubernetes/pkg/util/errors"
+)
+
+const (
+	sentinelFileName = "..sentinel"
+	tsFormat         = "1981_02_01_15_04_05"
+)
+
```

Comment:
Sorry Paul but really I'm not a fan of `func ts()`. It's used in on onlye one place (where you can inline the usual `time.Now().Format...` and the name it's not that great :). nit: but I would remove it